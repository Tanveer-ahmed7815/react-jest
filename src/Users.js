import React, { Component } from "react";

class Users extends Component{

    getUserList(){
        return "user list"
    }

    render(){
        return(
            <div>
                <h1>Users Class component </h1>
            </div>
        )
    }

}

export default Users;