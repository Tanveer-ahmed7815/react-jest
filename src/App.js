import logo from './logo.svg';
import './App.css';
import { useState } from 'react';
import Users from './Users';
import handleOtherMethod from './helper';

function App() {

  const [data,setData] =useState("");

 const handleData= () =>{
  setData("hello");
 }

  return (
    <div className="App">
     <p>First React test case</p>
     {/* <input type='text' value={data} onChange={(e)=>setData(e.target.value + "test")} placeholder='Enter username' /> */}
     <br />
     <br />
     {/* <button onClick={()=>setData("updated data")}>Update Data</button>
     <h1>{data}</h1> */}
     <br />
     <br />
     <Users />

      <h1>Functional component method testing</h1>
      <button data-testid="btn1" onClick={handleData}>Update</button>
      <button onClick={handleOtherMethod}>Print</button>
      <h2>{data}</h2>
 
      <br />
      <br />
        <h1>getByRole</h1>
        <input type='text'defaultValue="hello" disabled />
        <button>Click me</button>
        <div role='dummy'>
          dummy text
        </div>
      <br />
      <br />

      <h1>getAllByRole</h1>
     <button>Click me</button>
     <select>
      <option>1</option>
      <option>2</option>
      <option>3</option>
     </select>
     <br />
     <br />
     <label htmlFor='user-name1'>Username</label>
      <input type='text' id='user-name1' defaultValue={"Tanveer"}/>
      <br />
      <br />
      <label htmlFor='user-name2'>Username</label>
      <input type='text' id='user-name2' defaultValue={"Tanveer"}/>
      <br />
      <br />
      <label htmlFor='user-name3'>Username</label>
      <input type='text' id='user-name3' defaultValue={"Tanveer"}/>

      <br />
      <br />

      <input type='text' placeholder='enter username' />
      <input type='text' placeholder='enter username' />
      <input type='text' placeholder='enter username' />
      <input type='text' placeholder='enter username' />

      <br />
      <br />

      <button>Login</button>
      <p className='paragraph'>Login</p>
      <h1>Login</h1>

      <br />
      <br />
      <label htmlFor='skills'>Skills</label>
      <input type='checkbox' id='skills' defaultChecked={true}/>

      <br />
      <br />
     <img title='AI Genrated Image' src='https://image.jimcdn.com/app/cms/image/transf/none/path/sa6549607c78f5c11/image/i4eeacaa2dbf12d6d/version/1490299332/most-beautiful-landscapes-in-europe-lofoten-european-best-destinations-copyright-iakov-kalinin.jpg3919321_1443393332_n.jpg'
     />



    </div>
  );
}

export default App;
