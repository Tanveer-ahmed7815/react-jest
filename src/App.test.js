import { fireEvent, render, screen, configure, within, act } from "@testing-library/react";
import App from "./App";
import renderer from 'react-test-renderer';
import Users from "./Users";
import handleOtherMethod from "./helper";
import userEvent from "@testing-library/user-event";
configure({testIdAttribute:'element-id'})

beforeAll(() => {
  console.log("-------------- Before all hook -------------")
})

beforeEach(() => {
  console.log("--------------- Before each ---------------")
})

afterAll(() => {
  console.log("-------------- After all hook -------------")
})

afterEach(() => {
  console.log("--------------- After each ---------------")
})

describe.skip("UI test case group", () => {

  test("Test first React app case", () => {
    render(<App />);
    let text = screen.getByText(/First React test Case/i);
    const title = screen.getByTitle("AI Genrated Image");
    expect(text).toBeInTheDocument();
    expect(title).toBeInTheDocument();

  })

  test("Testing input box", () => {
    render(<App />);
    let checkInput = screen.getByRole("textbox");
    let checkInputPlaceholder = screen.getByPlaceholderText("Enter username");
    expect(checkInput).toBeInTheDocument();
    expect(checkInputPlaceholder).toBeInTheDocument();
  })

})

describe.skip("API test case group", () => {

  test("API Test first React app case", () => {
    render(<App />);
    let text = screen.getByText(/First React test Case/i);
    const title = screen.getByTitle("AI Genrated Image");
    expect(text).toBeInTheDocument();
    expect(title).toBeInTheDocument();

  })

  describe.skip("Nested decribe", () => {
    test("API Testing input box", () => {
      render(<App />);
      let checkInput = screen.getByRole("textbox");
      let checkInputPlaceholder = screen.getByPlaceholderText("Enter username");
      expect(checkInput).toBeInTheDocument();
      expect(checkInputPlaceholder).toBeInTheDocument();
    })
  })

})

describe.skip("On change event testing", () => {

  test("Onchange event testing", () => {
    render(<App />)
    let input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: 'a' } });
    expect(input.value).toBe("atest");
  })

})

describe.skip("On Click event testing for Button", () => {

  test("Onclick event testing for button", () => {
    render(<App />)
    let btn = screen.getByRole("button");
    fireEvent.click(btn);
    expect(screen.getByText("updated data")).toBeInTheDocument();
  })

})

describe.skip('Snapshot test cases', () => {
  test('Snapshot for App component', () => {

    const container = render(<App />);
    expect(container).toMatchSnapshot();

  })
})

describe.skip('Class component testing', () => {
  test('User class compponent testing', () => {

    const componentData = renderer.create(<Users />).getInstance();
    expect(componentData.getUserList()).toMatch("user list");

  })
})

describe.skip('Functional component testing', () => {
  test('Functional compponent method testing-1', () => {

    render(<App />)
    const btn = screen.getByTestId("btn1");
    fireEvent.click(btn);
    expect(screen.getByText("hello")).toBeInTheDocument();

  })

  test("Functional compponent method testing-1 for helper methods", () => {
    expect(handleOtherMethod()).toMatch("Hi");
  })
})

describe.skip("get by role", () => {
  test("getByRole testing", () => {
    render(<App />)
    const inputField = screen.getByRole("textbox");
    expect(inputField).toBeInTheDocument();
    expect(inputField).toHaveValue("hello");
    expect(inputField).toBeDisabled();
  })

  test("multiple role testing", () => {
    render(<App />)
    const btn1 = screen.getByRole("button", { name: "Click me" });
    expect(btn1).toBeInTheDocument();

  })

  test("non-semantic role testing", () => {
    render(<App />)
    const div1 = screen.getByRole("dummy");
    expect(div1).toBeInTheDocument();

  })
})

describe.skip('getAllByRole testing', () => {
  test('getAllByRole testing for button', () => {
    render(<App />);
    const btns = screen.getAllByRole("button");
    const options = screen.getAllByRole("option");

    for (let index = 0; index < btns.length; index++) {
      expect(btns[index]).toBeInTheDocument();
    }

    for (let index = 0; index < options.length; index++) {
      expect(options[index]).toBeInTheDocument();
    }
  })
})

describe.skip('getByLabelText/getAllByLabelText testing', () => {
  test('getByLabelText testing', () => {

    render(<App />);
    // const input = screen.getByLabelText("Username")
    // expect(input).toBeInTheDocument();
    // expect(input).toHaveValue("Tanveer")

  })

  test('getByLabelText testing for checkbox', () => {

    render(<App />);
    const checkbox = screen.getByLabelText("Skills")
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).toBeChecked();

  })

  test('getAllByLabelText testing', () => {

    render(<App />);
    const inputs = screen.getAllByLabelText("Username")
    expect(inputs[1]).toBeInTheDocument();

    for (let index = 0; index < inputs.length; index++) {
      expect(inputs[index]).toBeInTheDocument();
      expect(inputs[index]).toHaveValue("Tanveer")
      
    }

  })
})

describe.skip('getByPlaceholderText testing', () => {
  test('getByPlaceholderText testing', () => {
    render(<App />);
    // const inputs = screen.getByPlaceholderText("enter username")
    // expect(inputs).toBeInTheDocument();
    // expect(inputs).toHaveValue("Tanveer")

  })
  test('getByPlaceholderText testing', () => {
    render(<App />);
    const inputs = screen.getAllByPlaceholderText("enter username");
    
    for (let index = 0; index < inputs.length; index++) {
      expect(inputs[index]).toBeInTheDocument();
      
    }

  })
})

describe.skip('getByPlaceholderText testing', () => {
  test('getByPlaceholderText testing', () => {
    render(<App />);
    // const inputs = screen.getByText("Login")
    // expect(inputs).toBeInTheDocument();

  })
  test('getByPlaceholderText testing', () => {
    render(<App />);
    const inputs = screen.getAllByText("Login");
    
    for (let index = 0; index < inputs.length; index++) {
      expect(inputs[index]).toBeInTheDocument();
      
     }

  })
})

describe.skip('getByTestId/getAllByTestId testing', () => {
  test('getByTestId testing', () => {
    render(<App />);
    const inputs = screen.getByTestId("div-test-id")
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Overrriding test testing', () => {
  test('getByTestId testing', () => {
    render(<App />);
    const inputs = screen.getByTestId("div-test-id")
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing with display value', () => {
  test('getByDisplayValue testing', () => {
    render(<App />);
    const inputs = screen.getByDisplayValue("Ahmed")
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing with title', () => {
  test('getByTitle testing', () => {
    render(<App />);
    const inputs = screen.getByTitle("click me button")
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing with Alt text', () => {
  test('getByTitle testing', () => {
    render(<App />);
    const inputs = screen.getByAltText("dummy image")
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing for text match with String/regex', () => {
  test('text match testing with string', () => {
    render(<App />);
    const inputs = screen.getByText("Text ",{exact:false})
    expect(inputs).toBeInTheDocument();

  })

  test('text match testing with regex', () => {
    render(<App />);
    const inputs = screen.getByText(/text/)
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing for text match with function', () => {
  test('text match testing with function', () => {
    render(<App />);
    //const inputs = screen.getByText((content,element)=> content.startsWith("Text"));
    // const inputs = screen.getByText((content,element)=> content.endsWith("match"));
    const inputs = screen.getByText((content,element)=> {
      return content.includes("ex")
    });
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing for queryBy', () => {
  test('queryBy testing', () => {
    render(<App />);
    const inputs = screen.queryByText("Logout-hidden")
  
    expect(inputs).not.toBeInTheDocument();

  })
})

describe.skip('Testing for findBy',  () => {
  test('findBy testing', () => {
    render(<App />);
    const inputs = screen.findByText("data found",);
  
    expect(inputs).toBeInTheDocument();

  })
})

describe.skip('Testing for custome query',  () => {
  test('querySelector testing', () => {
    render(<App />);

    const element = document.querySelector('#testId')
    expect(element).toBeInTheDocument();

  })
})

describe.skip('Testing for Querying within elements',  () => {
  test('Query within elements testing', () => {
    render(<App />);

    const element = screen.getByText("Hello world");
    const subElement = within(element).getByText("Hi")
    expect(element).toBeInTheDocument();
    expect(subElement).toBeInTheDocument();

  })
})

describe.skip('Testing for click event with user event library',  () => {
  test('User event library testing', async () => {
    userEvent.setup();
    render(<App />);

    const btn = screen.getByText("Click me");
    await userEvent.click(btn);
    expect(screen.getByText("hello")).toBeInTheDocument();

  })
})

describe.skip('Testing for onchange event',  () => {
  test('Onchange event testing', async () => {
    userEvent.setup();
    render(<App />);

    const el = screen.getByRole("textbox");
    await userEvent.type(el,"Tanveer")
    expect(screen.getByText("Tanveer")).toBeInTheDocument();

  })
})

describe('Testing for Act function',  () => {
  test('Act function testing', async () => {
    userEvent.setup();
    render(<App />);

    const el = screen.getByRole("textbox");
    
    await act(async () =>{
      await userEvent.type(el,"Tanveer");
    })

    expect(screen.getByText("Tanveer")).toBeInTheDocument();

  })
})