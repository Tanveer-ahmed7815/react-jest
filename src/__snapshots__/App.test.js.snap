// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test cases Snapshot for App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <p>
          First React test case
        </p>
        <input
          placeholder="Enter username"
          type="text"
          value=""
        />
        <br />
        <br />
        <button>
          Update Data
        </button>
        <h1 />
        <br />
        <br />
        <img
          src="https://image.jimcdn.com/app/cms/image/transf/none/path/sa6549607c78f5c11/image/i4eeacaa2dbf12d6d/version/1490299332/most-beautiful-landscapes-in-europe-lofoten-european-best-destinations-copyright-iakov-kalinin.jpg3919321_1443393332_n.jpg"
          title="AI Genrated Image"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <p>
        First React test case
      </p>
      <input
        placeholder="Enter username"
        type="text"
        value=""
      />
      <br />
      <br />
      <button>
        Update Data
      </button>
      <h1 />
      <br />
      <br />
      <img
        src="https://image.jimcdn.com/app/cms/image/transf/none/path/sa6549607c78f5c11/image/i4eeacaa2dbf12d6d/version/1490299332/most-beautiful-landscapes-in-europe-lofoten-european-best-destinations-copyright-iakov-kalinin.jpg3919321_1443393332_n.jpg"
        title="AI Genrated Image"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
